src/init.cpp

```diff
@@ -1871,9 +1871,12 @@ bool AppInitMain()
 
     g_wallet_init_interface.Start(scheduler);
 
+#ifdef ENABLE_WALLET
```

Comment:
This is preprocessor, so only changes the way the wallet is compiled. Whereas `-disablewallet` is a runtime option and won't be able to take advantage of this change. I would suggest checking for the startup arg directly:
`gArgs.GetBoolArg("-disablewallet", DEFAULT_DISABLE_WALLET)` 
The above code would return true if the wallet is disabled.